src/lib.rs

```diff
@@ -59,6 +60,53 @@
 //! }
 //! ```
 //!
+//! Example usage with TLS:
+//!
+#![cfg_attr(feature = "tls", doc = " ```rust,no_run")]
```

Comment:
These provide some configurability for what runs during rustdoc's test.

So for these two:

```rust
#![cfg_attr(feature = "tls", doc = " ```ignore")]
#![cfg_attr(not(feature = "tls"), doc = " ```")]
```

It's interpreted something like this:

```
if feature != tls { return //! ``` };
if feature == tls { return //! ```ignore };
```

`ignore` just means don't run this test case at all. `no_run` is going to _compile_ the test, but not actually run it. We want this for the tls use case because we need to extra stuff like in the test case to add the certs to the cert chain, which we don't need to here in the rustdoc.

For the tcp side actually we can run that now even with tls enabled, so the first set of conditions can be removed.